backend/api/orders/mutations.py

```diff
@@ -38,3 +55,44 @@ def create_order(
         payment_url += f"?return_url={return_url}"
 
         return CreateOrderResult(payment_url=payment_url)
+
+
+def validate_hotel_rooms(hotel_rooms: typing.List[CreateOrderHotelRoom], *, conference):
+    for order_room in hotel_rooms:
+        try:
+            room = HotelRoom.objects.get(id=order_room.room_id)
+        except HotelRoom.DoesNotExist:
+            return Error(
+                message=_("Room %(id)s not found") & {"id": order_room.room_id}
+            )
+
+        if room.is_sold_out:
+            return Error(
+                message=_("Room %(id)s is sold out") & {"id": order_room.room_id}
+            )
+
+        conference_start = conference.start.date()
+        conference_end = conference.end.date()
+
+        if order_room.checkin < conference_start or order_room.checkin > conference_end:
+            return Error(message=_("Invaild checkin date"))
+
+        if (
+            order_room.checkout < conference_start
+            or order_room.checkout > conference_end
+            or order_room.checkin > order_room.checkout
+        ):
+            return Error(message=_("Invaild checkout date"))
+
+
+def create_hotel_reservation(
```

Comment:
I'd rename this to create_hotel_reservations (plural) since it creates multiple reservations :)

Reply:
done!